File: 0001-two-sum/0001-two-sum.java
class Solution {
    public int[] twoSum(int[] nums, int target) {
        HashMap<Integer, Integer> map = new HashMap();
        
        for(int i = 0; i < nums.length;i++)
        {
            int current = nums[i];
            int remaining = target - current;
            
            if(map.containsKey(remaining))
                return new int[]{i, map.get(remaining)};
            
            map.put(current, i);
        }
        
        return new int[]{-1,-1};
    }
}

File: 0001-two-sum/0001-two-sum.py
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        remainingMap = dict()

        for i,number in enumerate(nums):
            remaining = target - number
            if(remaining in remainingMap):
                return [remainingMap[remaining],i]
            remainingMap[number] = i
        return []